packages/connect/crates/connect-relay/src/server.rs

```diff
@@ -0,0 +1,889 @@
+use std::{collections::HashMap, pin::Pin};
+
+use crate::constants::DEFAULT_RELAYED_CONNECTION_PING_TIMEOUT;
+use futures::{
+    channel::mpsc::{self, UnboundedReceiver, UnboundedSender},
+    future::{select, Either},
+    pin_mut,
+    stream::{FusedStream, Stream},
+    task::{Context, Poll},
+    Future, FutureExt, Sink, SinkExt, TryFutureExt,
+};
+use getrandom::getrandom;
+use pin_project_lite::pin_project;
+use std::task::Waker;
+use utils_log::{error, info};
+
+#[cfg(feature = "wasm")]
+use crate::streaming_iterable::{JsStreamingIterable, StreamingIterable};
+
+#[cfg(feature = "wasm")]
+use gloo_timers::future::sleep;
+#[cfg(not(feature = "wasm"))]
+use utils_misc::time::native::sleep;
+
+#[cfg(not(feature = "wasm"))]
+use utils_misc::time::native::current_timestamp;
+#[cfg(feature = "wasm")]
+use utils_misc::time::wasm::current_timestamp;
+
+#[repr(u8)]
+pub enum MessagePrefix {
+    Payload = 0x00,
+    StatusMessage = 0x01,
+    ConnectionStatus = 0x02,
+    WebRTC = 0x03,
+}
+
+#[repr(u8)]
+pub enum StatusMessage {
+    Ping = 0x00,
+    Pong = 0x01,
+}
+
+#[repr(u8)]
+pub enum ConnectionStatusMessage {
+    Stop = 0x00,
+    Restart = 0x01,
+    Upgraded = 0x02,
+}
+
+#[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
+extern "C" {
+    #[wasm_bindgen]
+    #[derive(Clone, Debug)]
+    pub type RelayServerCbs;
+
+    #[wasm_bindgen(js_name = "onClose", structural, method)]
+    pub fn on_close(this: &RelayServerCbs);
+
+    #[wasm_bindgen(js_name = "onUpgrade", structural, method)]
+    pub fn on_upgrade(this: &RelayServerCbs);
+}
+
```

Comment:
Please add at least empty implementation (with methods performing `unimplemented!("...")` panic) of `RelayServerCbs` so this code at least builds with the WASM feature disabled.